finality-aleph/src/tcp_network.rs

```diff
@@ -10,9 +10,36 @@ use tokio::net::{
 
 use crate::{
     network::{Multiaddress, NetworkIdentity, PeerId},
-    validator_network::{Dialer, Listener, Splittable},
+    validator_network::{ConnectionInfo, Dialer, Listener, Splittable},
 };
 
+impl ConnectionInfo for TcpStream {
+    fn peer_address_info(&self) -> String {
+        match self.peer_addr() {
+            Ok(addr) => addr.to_string(),
+            Err(e) => e.to_string(),
```

Comment:
I would suggest formatting the error more explicitly as an error here, as in "unknown address: {}" or something.

Reply:
fixed